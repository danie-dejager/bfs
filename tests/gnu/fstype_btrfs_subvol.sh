# Test that -fstype works in btrfs subvolumes

command -v btrfs &>/dev/null || skip

cd "$TEST"

# Make a btrfs filesystem image
truncate -s128M img
mkfs.btrfs img >&2

# Mount it
mkdir mnt
bfs_sudo mount img mnt || skip
defer bfs_sudo umount mnt

# Make it owned by us
bfs_sudo chown "$(id -u):$(id -g)" mnt

# Create a subvolume inside it
btrfs subvolume create mnt/subvol >&2

# Make a file in and outside the subvolume
"$XTOUCH" mnt/file mnt/subvol/file

bfs_diff mnt -fstype btrfs -print -o -printf '%p %F\n'
