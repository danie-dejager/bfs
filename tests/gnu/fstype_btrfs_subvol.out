mnt
mnt/file
mnt/subvol
mnt/subvol/file
